### ISSUE TITLE
Integrate `NNStopping` and `NNKolmogorov ` in HighDimPDE.jl

### ISSUE THREAD

**Issue by vboussange** [SciML/NeuralPDE.jl#536]
`NeuralPDE.KolmogorovPDEProblem`, documented in 
- https://neuralpde.sciml.ai/dev/solvers/kolmogorovbackwards_solver/
- https://neuralpde.sciml.ai/dev/examples/kolmogorovbackwards/

are `HighDimPDE.PIDEProblem` where the nonlinear term is 0. 
Associated code and documentation should be migrated to HighDimPDE.jl, which is a more appropriate package to host those types of problem.

Similarly, the optimal stopping problem solver `NeuralPDE.NNStopping`, documented in 
- https://neuralpde.sciml.ai/dev/solvers/optimal_stopping/
- https://neuralpde.sciml.ai/dev/examples/optimal_stopping_american/

should be transferred to HighDimPDE.jl.

**Comment by KirillZubov** [comment]
@vboussange which can I help with it issue?

**Comment by vboussange** [comment]
@KirillZubov yes, it would be great! I have not started any of those. You could try to implement `KolmogorovPDEProblem` and `NNStopping` on the branch https://github.com/SciML/HighDimPDE.jl/tree/NeuralPDE_integration, as well as transferring the documentation.

**Comment by KirillZubov** [comment]
ok, I will do it
